### PR TITLE
feat: code review

### PR DESCRIPTION
**Pull request by RealShadowNova**

### CODE REVIEW

File: packages/provider/src/lib/structures/JoshMiddlewareStore.ts [hunk 1]
```diff
@@ -45,13 +45,15 @@ export class JoshMiddlewareStore<StoredValue = unknown> extends Map<string, Josh
   /**
    * Filter middlewares by their conditions.
    * @since 1.0.0
-   * @param method
-   * @param trigger
+   * @param method The method to filter by.
+   * @param trigger The filtered middlewares.
```
Comment (dan-online) [inline packages/provider/src/lib/structures/JoshMiddlewareStore.ts:49]: How is the trigger filtered middleware?

File: packages/provider/src/lib/types/Payload.ts [hunk 2]
```diff
@@ -157,5 +152,507 @@ export namespace Payload {
...
   }
 
+  /**
+   * The hook type for {@link Payload}
+   * @since 1.0.0
+   */
   export type Hook<Value, R = unknown> = (value: Value, key: string) => Awaitable<R>;
+
+  /**
+   * The payload for {@link Method.Dec}
```
Comment (dan-online) [inline packages/provider/src/lib/types/Payload.ts:162]: AutoKey? Not Dec

File: packages/provider/src/lib/types/Payload.ts [hunk 3]
```diff
@@ -157,5 +152,507 @@ export namespace Payload {
...
+  /**
+   * The payload for {@link Method.Delete}
+   * @since 1.0.0
+   */
+  export interface Delete extends Payload, KeyPath {
+    method: Method.Delete;
+  }
+
+  /**
+   * The payload for {@link Method.Get}
```
Comment (dan-online) [inline packages/provider/src/lib/types/Payload.ts:194]: DeleteMany?

File: packages/provider/src/lib/types/Payload.ts [hunk 4]
```diff
@@ -157,5 +152,507 @@ export namespace Payload {
...
+  }
+
+  export namespace Each {
+    export interface Metadata extends Record<PropertyKey, unknown> {
+      skipProvider?: boolean;
+    }
+  }
+
+  /**
+   * The payload for {@link Method.Get}
```
Comment (dan-online) [inline packages/provider/src/lib/types/Payload.ts:230]: Ensure??

File: packages/provider/src/lib/types/Payload.ts [hunk 5]
```diff
@@ -157,5 +152,507 @@ export namespace Payload {
...
+    overwrite: boolean;
+
+    /**
+     * The entries to set.
+     * @since 1.0.0
+     */
+    entries: SetMany.KeyPathValue[];
+  }
+
+  export namespace SetMany {
```
Comment (dan-online) [inline packages/provider/src/lib/types/Payload.ts:593]: Description needed?

File: packages/serialize/README.md [hunk 6]
```diff
@@ -16,12 +16,12 @@
 
 ## Description
 
-A serialize utility to be able to convert nodejs data types into a json compatible format.
+Do you want serialize Node.js data types to a JSON compatible format? Then this is the package for you!
```
Comment (dan-online) [inline packages/serialize/README.md:19]: Lad, maybe something a bit more descriptive haha

**Review by dan-online — CHANGES_REQUESTED** [review]

File: packages/provider/src/lib/types/Payload.ts (hunk 5 above)
Comment (RealShadowNova) [reply packages/provider/src/lib/types/Payload.ts:593]: No, this is the namespace, see the interface.

File: packages/serialize/README.md (hunk 6 above)
Comment (RealShadowNova) [reply packages/serialize/README.md:19]: Not sure how much more descriptive it gets lmao

**Review by dan-online — APPROVED** [review]

**Pull request merged**